java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Field.java

```diff
@@ -47,6 +49,8 @@
 
 public class Field {
 
+  public static final String ZEROVECTOR_OLD_NAME = "[DEFAULT]";
```

Comment:
Just move this directly into `mutateOriginalNameIfNeeded` and same with the tests, no need to make this a static variable in Field.

Reply:
mutateOriginalNameIfNeeded may be invoked many times. I understand that compiler will optimize it, but I am not sure it makes sense to multiply "[DEFAULT]". May be place to keep this const is not a great one, I would prefer to keep it in ZeroVector - at least it has some association with it.